### PR TITLE
Refactoring ships

### PR DESCRIPTION
**Pull request by PixelDotPixel**
Ra factored ships and added a submarine class

### CODE REVIEW

File: src/main/java/cs361/battleships/models/Board.java [hunk 1]
```diff
@@ -34,7 +36,19 @@ public boolean placeShip(Ship ship, int x, char y, boolean isVertical) {
 		if (ships.stream().anyMatch(s -> s.getKind().equals(ship.getKind()))) {
 			return false;
 		}
-		final var placedShip = new Ship(ship.getKind());
+
+		var placedShipT = new Ship();
```
Comment (amayaevan) [inline src/main/java/cs361/battleships/models/Board.java]: I believe the new submarine ship should be added to this switch statement.

File: src/main/java/cs361/battleships/models/Ship.java [hunk 2]
```diff
@@ -42,41 +46,11 @@ public Ship(String kind) {
 	}
 
 	public void place(char col, int row, boolean isVertical) {
-		int CommandLocation = -1;
-		int ArmourPoints = 0;
-		switch(kind) {
```
Comment (amayaevan) [inline src/main/java/cs361/battleships/models/Ship.java]: This is good in terms of making the bloated ship class smaller and making it more generic to be capable of being expanded on in subclasses.

File: src/main/java/cs361/battleships/models/Ship_Submarine.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import jdk.jshell.Snippet;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+public class Ship_Submarine extends Ship_CaptainsQuarters {
+
+    public Ship_Submarine() {
```
Comment (amayaevan) [inline src/main/java/cs361/battleships/models/Ship_Submarine.java]: This is a good start on the submarine ship type. This ship type needs to be added to the ship/board tests in order to fully test implementation and make sure it functions properly with the other mechanics of the game.

We already discussed this part but I just wanted to keep it as a note

**Comment by zhangji52** [comment]
It definitely saves a lot of space for the codes, instead of naming the type of ship every time by saying var placedShipT = new Ship();, so that we don't have to define or declare those. It also helped the other team members to know the class of ship_submarine, easier to see it. Good work on the debugging!

**Comment by IsaacPeters** [comment]
I think you probably could have cut down on some of the test cases, but the refactoring looks good.

**Pull request merged**